packages/client/tests/functional/tracing/tests.ts

```diff
@@ -84,308 +84,378 @@ afterAll(() => {
   context.disable()
 })
 
-testMatrix.setupTestSuite(
-  ({ provider }, suiteMeta, clientMeta) => {
-    const isMongoDb = provider === Providers.MONGODB
+testMatrix.setupTestSuite(({ provider, providerFlavor, relationMode }, _suiteMeta, clientMeta) => {
+  const isMongoDb = provider === Providers.MONGODB
 
-    beforeEach(async () => {
-      await prisma.$connect()
-      inMemorySpanExporter.reset()
+  const usesSyntheticTxQueries =
+    providerFlavor !== undefined && ['js_libsql', 'js_planetscale'].includes(providerFlavor)
+
+  beforeEach(async () => {
+    await prisma.$connect()
+    inMemorySpanExporter.reset()
+  })
+
+  async function waitForSpanTree(expectedTree: Tree): Promise<void> {
+    await waitFor(async () => {
+      await processor.forceFlush()
+      const spans = inMemorySpanExporter.getFinishedSpans()
+      const rootSpan = spans.find((span) => !span.parentSpanId) as ReadableSpan
+      const tree = buildTree(rootSpan, spans)
+
+      expect(tree).toEqual(expectedTree)
     })
+  }
 
-    async function waitForSpanTree(expectedTree: Tree): Promise<void> {
-      await waitFor(async () => {
-        await processor.forceFlush()
-        const spans = inMemorySpanExporter.getFinishedSpans()
-        const rootSpan = spans.find((span) => !span.parentSpanId) as ReadableSpan
-        const tree = buildTree(rootSpan, spans)
+  enum QueryChildSpans {
```

Comment:
Maybe this suggestion could make it more explicit?
```suggestion
  enum AdapterQueryChildSpans {
```